code/modules/atmospherics/machinery/portable/canister.dm

```diff
@@ -451,10 +456,16 @@
 /obj/machinery/portable_atmospherics/canister/proc/canister_break()
 	disconnect()
 	var/datum/gas_mixture/expelled_gas = air_contents.remove(air_contents.total_moles())
+	var/expelled_pressure = expelled_gas?.return_pressure()
 	var/turf/T = get_turf(src)
 	T.assume_air(expelled_gas)
 	air_update_turf(FALSE, FALSE)
 
+	if(expelled_pressure > pressure_limit)
+		var/pressure_dif = expelled_pressure - pressure_limit
+		var/explosion_range = CEILING(min(pressure_dif, 20000) / 1000, 1)
+		explosion(T, 0, 0, explosion_range, 0, smoke = FALSE)
```

Comment:
I'd move this to after the call to obj_break() so it doesn't somehow get deleted by the explosion consequences

Reply:
I've made the break happen first, though since this is only light damage I don't think it would have be able to delete it in the first place.